### PR TITLE
CIApp: Correctness and stability fixes.

### PR DESCRIPTION
**Pull request by tonyredondo**
- Fixes a scenario when a NUnit test suite fails on the OneTimeSetUp method gets reported different than the console output.
- Improves render of parameter values.

@DataDog/apm-dotnet

### CODE REVIEW

File: src/Datadog.Trace.ClrProfiler.Managed/AutoInstrumentation/Testing/NUnit/IResultState.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        /// Gets the TestStatus for the test.
+        /// </summary>
+        /// <value>The status.</value>
+        TestStatus Status { get; }
+
+        /// <summary>
+        /// Gets the label under which this test result is
+        /// categorized, or <see cref="string.Empty"/> if none.
+        /// </summary>
+        string Label { get; }
```
Comment (andrewlock) [inline src/Datadog.Trace.ClrProfiler.Managed/AutoInstrumentation/Testing/NUnit/IResultState.cs:23]: I don't think this property is used, do you need it?

**Review by andrewlock — APPROVED** [review]

File: src/Datadog.Trace.ClrProfiler.Managed/AutoInstrumentation/Testing/NUnit/IResultState.cs (hunk 1 above)
Comment (tonyredondo) [reply src/Datadog.Trace.ClrProfiler.Managed/AutoInstrumentation/Testing/NUnit/IResultState.cs:23]: you're right, removing it.

Comment (tonyredondo) [reply src/Datadog.Trace.ClrProfiler.Managed/AutoInstrumentation/Testing/NUnit/IResultState.cs]: done in fd3279a

File: src/Datadog.Trace.ClrProfiler.Managed/AutoInstrumentation/Testing/NUnit/NUnitCompositeWorkItemSkipChildrenIntegration.cs [hunk 2]
```diff
@@ -36,6 +40,60 @@ public class NUnitCompositeWorkItemSkipChildrenIntegration
...
+                        {
+                            if (item.GetType().Name == "CompositeWorkItem")
+                            {
+                                var compositeWorkItem2 = item.DuckCast<ICompositeWorkItem>();
+                                foreach (var item2 in compositeWorkItem2.Children)
+                                {
+                                    var testResult = item2.DuckCast<IWorkItem>().Result;
+                                    Scope scope = NUnitIntegration.CreateScope(testResult.Test, typeof(TTarget));
+                                    scope.Span.Error = true;
+                                    scope.Span.SetTag(Tags.ErrorMsg, compositeWorkItem.Result.Message);
```
Comment (zacharycmontoya) [inline src/Datadog.Trace.ClrProfiler.Managed/AutoInstrumentation/Testing/NUnit/NUnitCompositeWorkItemSkipChildrenIntegration.cs:64]: Just to double check, did you intend for these tags to come from `compositeWorkItem` or did you mean to use `compositeWorkItem2`?

Comment (tonyredondo) [reply src/Datadog.Trace.ClrProfiler.Managed/AutoInstrumentation/Testing/NUnit/NUnitCompositeWorkItemSkipChildrenIntegration.cs:64]: yes, the first `compositeWorkItem` contains the actual exception from the `OneTimeSetup` method, so what I do is to loop over the children and tag with the same exception from the setup method.

**Review by zacharycmontoya — APPROVED** [review]

**Pull request merged**